linera-core/src/local_node.rs

```diff
@@ -110,29 +110,34 @@ where
         certificate: LiteCertificate<'_>,
         notifier: &impl Notifier,
     ) -> Result<ChainInfoResponse, LocalNodeError> {
-        let full_cert = self.node.state.full_certificate(certificate).await?;
-        let response = self
-            .node
-            .state
-            .fully_handle_certificate_with_notifications(full_cert, vec![], notifier)
-            .await?;
-        Ok(response)
+        match self.node.state.full_certificate(certificate).await? {
+            Either::Left(confirmed) => {
+                Ok(self.handle_certificate(confirmed, vec![], notifier).await?)
+            }
+            Either::Right(validated) => {
+                Ok(self.handle_certificate(validated, vec![], notifier).await?)
+            }
+        }
     }
 
     #[instrument(level = "trace", skip_all)]
-    pub async fn handle_certificate(
+    pub async fn handle_certificate<T>(
         &self,
-        certificate: Certificate,
+        certificate: GenericCertificate<T>,
         blobs: Vec<Blob>,
         notifier: &impl Notifier,
-    ) -> Result<ChainInfoResponse, LocalNodeError> {
-        let response = Box::pin(self.node.state.fully_handle_certificate_with_notifications(
-            certificate,
-            blobs,
-            notifier,
-        ))
-        .await?;
-        Ok(response)
+    ) -> Result<ChainInfoResponse, LocalNodeError>
+    where
+        T: CertificateProcessor,
+    {
+        Ok(
+            Box::pin(self.node.state.fully_handle_certificate_with_notifications(
```

Comment:
Is the `pin` still necessary here? Looks like it shouldn't be to me...

Reply:
Whoops, looks like it was needed after all :grimacing: (CI threw some errors because of its removal.)

Reply:
Yeah... brought back.